desktop/components/react/main_layout/layout.jsx

```diff
@@ -0,0 +1,138 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import capitalize from 'underscore.string/capitalize'
+import getLayout from './utils/get_layout'
+
+/**
+ * Utility for working with React and Jade layout code on the server. If raw
+ * template html is passed in as props it will return renderable components for
+ * use in Express jsx views.
+ *
+ * @example
+ *
+ * // routes.js
+ * import templateRenderer from 'components/react/utils/template_renderer'
+ *
+ * const [meta] = templateRenderer(['meta.jade']).render(res.locals)
+ *
+ * res.render('index.jsx', {
+ *   templates: {
+ *     meta // html string
+ *   }
+ * })
+ *
+ * // templates/index.jsx
+ * import { makeLayout } from 'components/react/main_layout'
+ *
+ * export default IndexRoute (props) {
+ *   const { Layout: { Header, Body, Footer }, Meta } = makeLayout(props)
+ *
+ *   return (
+ *     <div>
+ *       <Header>
+ *         <Meta />
+ *       </Header>
+ *       <Body>
+ *         ...
+ *       </Body>
+ *       <Footer />
+ *     </div>
+ *   )
+ * }
+ *
+ * @param  {Object} [templates={}] Optional templates to map into React components
+ * @param  {Object} props          Data to render into components
+ * @return {Object}
+ */
+export function makeLayout ({ templates, ...props } = { templates: {} }) {
```

Comment:
We bounce a little bit between React's `props` nomenclature and Express's `locals` nomenclature, which is a little confusing. These are really just `locals` right?